packages/webapp/src/app/cost/page.tsx

```diff
@@ -90,12 +90,21 @@ export default async function CostAnalysisPage({
   }
 
   // Group data by different dimensions
-  const sessionCosts = filteredSessions.map((session) => ({
-    workerId: session.workerId,
-    initialMessage: session.initialMessage,
-    sessionCost: session.sessionCost || 0,
-    createdAt: session.createdAt,
-  }));
+  const sessionCosts = await Promise.all(
+    filteredSessions.map(async (session) => {
+      // Get repository metadata for this session
+      const repoMetadata = await readMetadata('repo', session.workerId);
```

Comment:
N+1 problem. We'll optimize this later.

(Or maybe it won't be problem since #sessions is often less than 1000 and we shouldn't open the cost page very frequently.)